test-suite-kotlin-ksp/src/test/kotlin/io/micronaut/reproduce/reproduce.kt

```diff
@@ -0,0 +1,20 @@
+package io.micronaut.reproduce
+
+import io.micronaut.context.ApplicationContext
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class ReproduceIssueSpec {
+
+    @Test
+    fun `Kotlin child inherits field from Java parent with KSP`() {
+        val ctx = ApplicationContext.run()
+
+        // Correctly reference the class for Micronaut to create a bean
+        assertThrows<IllegalStateException> {
```

Comment:
@Jarvx  If you fix this bug, Why do you expect an error? It turns out that after your fix there shouldn't be an error.